### PR TITLE
[WIP] Update to the latest backend

### PR DESCRIPTION
**Pull request by erbridge**
I've commented out the theme code to get it building, but we should fix it instead. @zoli: How are themes / colour schemes supposed to work now?

### CODE REVIEW

**Comment by zoli** [comment]
The textmate package now lives here https://github.com/limetext/sublime/tree/master/internal/textmate and should not be used from outside sublime package.

> How are themes / colour schemes supposed to work now?

Every color scheme and syntax should implement [`backend.ColorScheme`](https://github.com/limetext/backend/blob/master/view.go#L57) and [`backend.Syntax`](https://github.com/limetext/backend/blob/master/view.go#L64) interface so they could be registered to backend via [`editor.AddColorScheme`](https://github.com/limetext/backend/blob/master/editor.go#L472) or [`editor.AddSyntax`](https://github.com/limetext/backend/blob/master/editor.go#L483).
Currently sublime package does [this](https://github.com/limetext/sublime/blob/master/package.go#L127-L150) it adds every theme and syntax that are loaded to backend.

**Comment by zoli** [comment]
This new implementation makes us to wait for loading all packages first after that we are able to get the color scheme.

We should have a default syntax and color scheme in the backend not depending on any package to resolve this issue. Then loading packages can be done concurrently in separate go routine.

**Comment by erbridge** [comment]
Ok, and then you call `editor.GetColorScheme` to get the actual scheme?

**Comment by zoli** [comment]
Yes something like:

``` go
ed.GetColorScheme(ed.Settings().Get("color_scheme", "").(string))
```

File: main/main.go [hunk 1]
```diff
@@ -158,15 +157,10 @@ func createFrontend() *tbfe {
...
+	t.console.AddObserver(&t)
 	t.setupCallbacks(t.currentView)
 
-	path := path.Join(backend.LIME_PACKAGES_PATH, "themes", "TextMate-Themes", "Monokai.tmTheme")
-	if sc, err := textmate.LoadTheme(path); err != nil {
-		log.Error(err)
-	} else {
-		scheme = sc
-	}
+	t.editor.Settings().Set("color_scheme", path.Join("..", "packages", "themes", "TextMate-Themes", "Monokai.tmTheme"))
```
Comment (zoli) [inline main/main.go]: We have set color scheme in [default settings](https://github.com/limetext/lime-packages/blob/master/Default/Preferences.sublime-settings#L27) so no need to set it manually here.

File: main/main.go [hunk 2]
```diff
@@ -760,32 +754,33 @@ func setColorMode() {
...
+	// 	}
+	// }
 }
 
 func createNewView(filename string, window *backend.Window) *backend.View {
 	v := window.OpenFile(filename, 0)
 
 	v.Settings().Set("trace", true)
-	v.SetSyntaxFile(path.Join(backend.LIME_PACKAGES_PATH, "go.tmbundle", "Syntaxes", "Go.tmLanguage"))
+	v.SetSyntaxFile(path.Join("..", "packages", "go-tmbundle", "Syntaxes", "Go.tmLanguage"))
```
Comment (zoli) [inline main/main.go]: Also we have syntax detection based on file type from https://github.com/limetext/backend/commit/9d7f5428022bda8c896f2412d00394a5e31cfe61, so no need to set the syntax manually.

**Comment by zoli** [comment]
The travis is failing because there should be [sublime_plugin.py](https://github.com/limetext/lime-qml/blob/master/main/sublime_plugin.py) file included in root directory of executable file(we have it in lime-qml/main to).
Instead of logging it as error its failing we have and issue for it https://github.com/limetext/sublime/issues/2.

File: main/main.go (hunk 2 above)
Comment (erbridge) [inline main/main.go]: Nice :).

**Comment by erbridge** [comment]
Copying in `sublime_plugin.py` is really nasty. We should add a build step that does that from the actual package.

**Comment by zoli** [comment]
Yes we should add build step in makefile so it copies `sublime_plugin.py` from sublime package

**Comment by erbridge** [comment]
Yup.

I'm going to merge this in and look at fixing whatever `setSchemeSettings` was supposed to do some other time. Ok with that @zoli?

**Comment by erbridge** [comment]
Resolves #11.

**Comment by zoli** [comment]
LGTM, nice to see termbox frontend working again.

**Comment by zoli** [comment]
A note on `setSchemeSettings` after getting the color scheme from editor we can access global settings via [`Global()`](https://github.com/limetext/backend/blob/master/render/renderer.go#L44-L45) so STH like `s.Settings["foreground"]` should change to `scheme.Global().Foreground`.

All available global settings are mentioned [here](https://github.com/limetext/backend/blob/master/render/settings.go).

**Comment by erbridge** [comment]
Ah great. Thanks.

**Pull request merged**